### ISSUE TITLE
ERROR - The testing results of the whole dataset is empty.

### ISSUE THREAD

**Issue by HaoweiGis** [open-mmlab/mmdetection#3269]
**Describe the bug**
When I run faster_rcnn_r101_fpn_1x_coco, I can get the bbox evaluation, but I run faster_rcnn_hrnetv2p_w40_1x_coco, faster_rcnn_x101_64x4d_fpn_1x_coco I got the error. `ERROR - The testing results of the whole dataset is empty.` Use tools/test.py to get the same error

**Environment**
1. `python tools/train.py configs/faster_rcnn/faster_rcnn_r101_fpn_1x_coco.py` is OK.
2. `python tools/train.py configs/faster_rcnn/faster_rcnn_x101_64x4d_fpn_1x_coco.py` is ERROR.
3. `python tools/train.py configs/hrnet/faster_rcnn_hrnetv2p_w40_1x_coco.py` is ERROR.
I use Tesla V100-PCIE-32GB, The three models are trained simultaneously！

I need help！

**Comment by xvjiarui** [comment]
Hi @HaoweiGis 
How many gpus did you use to train?

**Comment by HaoweiGis** [comment]
> Hi @HaoweiGis
> How many gpus did you use to train?

Hello, I am training multiple models on a V100 GPU

**Comment by HaoweiGis** [comment]
After I use this tutorial，[https://github.com/open-mmlab/mmdetection/issues/2942](url)

 `python tools/train.py configs/faster_rcnn/faster_rcnn_x101_64x4d_fpn_1x_coco.py` is OK. But faster_rcnn_hrnetv2p_w40_1x_coco still does not work, I am more confused about this.

**Comment by HaoweiGis** [comment]
After several iterations of the model, the loss becomes Nan
![image](https://user-images.githubusercontent.com/29857866/87220581-a097c380-c397-11ea-91d3-f7cc67185c61.png)

@xvjiarui 
unexpected key in source state_dict: incre_modules.0.0.conv1.weight, incre_modules.0.0.bn1.weight... ,是不是和这个有关系呢

**Comment by farshid-mi** [comment]
hi,same thing here for **faster_rcnn_r50_fpn_1x_coco** on **colab** training on **GWHD** dataset in shape of coco_instance,resulting in :

> 2020-07-13 08:35:03,306 - mmdet - INFO - Epoch [7][1200/1501]	lr: 2.000e-02, eta: 1:05:02, time: 0.495, data_time: 0.049, memory: 927, loss_rpn_cls: 0.5727, loss_rpn_bbox: 0.2353, loss_cls: 0.3850, acc: 86.2793, loss_bbox: 0.3781, loss: 1.5711

> 2020-07-13 08:35:28,481 - mmdet - INFO - Epoch [7][1250/1501]	lr: 2.000e-02, eta: 1:04:37, time: 0.504, data_time: 0.049, memory: 927, loss_rpn_cls: 0.5864, loss_rpn_bbox: 0.2476, loss_cls: 0.4269, acc: 83.7559, loss_bbox: 0.3990, loss: 1.6599

> 2020-07-13 08:35:49,794 - mmdet - INFO - Epoch [7][1300/1501]	lr: 2.000e-02, eta: 1:04:09, time: 0.427, data_time: 0.048, memory: 927, loss_rpn_cls: **nan**, loss_rpn_bbox: **nan**, loss_cls: **nan**, acc: 96.1270, loss_bbox: **nan**, loss: **nan**

> 2020-07-13 08:36:10,318 - mmdet - INFO - Epoch [7][1350/1501]	lr: 2.000e-02, eta: 1:03:41, time: 0.410, data_time: 0.049, memory: 927, loss_rpn_cls: nan, loss_rpn_bbox: nan, loss_cls: nan, acc: 100.0000, loss_bbox: nan, loss: nan

> 2020-07-13 08:36:30,839 - mmdet - INFO - Epoch [7][1400/1501]	lr: 2.000e-02, eta: 1:03:13, time: 0.410, data_time: 0.047, memory: 927, loss_rpn_cls: nan, loss_rpn_bbox: nan, loss_cls: nan, acc: 100.0000, loss_bbox: nan, loss: nan

> 2020-07-13 08:36:51,602 - mmdet - INFO - Epoch [7][1450/1501]	lr: 2.000e-02, eta: 1:02:45, time: 0.415, data_time: 0.046, memory: 927, loss_rpn_cls: nan, loss_rpn_bbox: nan, loss_cls: nan, acc: 100.0000, loss_bbox: nan, loss: nan

> 2020-07-13 08:37:12,327 - mmdet - INFO - Epoch [7][1500/1501]	lr: 2.000e-02, eta: 1:02:17, time: 0.414, data_time: 0.047, memory: 927, loss_rpn_cls: nan, loss_rpn_bbox: nan, loss_cls: nan, acc: 100.0000, loss_bbox: nan, loss: nan

> 2020-07-13 08:37:12,604 - mmdet - INFO - Saving checkpoint at 7 epochs
> [>>] 39/39, 4.3 task/s, elapsed: 9s, ETA:     0s2020-07-13 08:37:22,439 - mmdet - INFO - Evaluating bbox...
> Loading and preparing results...
> 2020-07-13 08:37:22,440 - mmdet - ERROR - The testing results of the whole dataset is empty.

> 2020-07-13 08:37:22,440 - mmdet - INFO - Epoch [7][1501/1501]	lr: 2.000e-02, 

> 2020-07-13 08:37:44,977 - mmdet - INFO - Epoch [8][50/1501]	lr: 2.000e-02, eta: 1:01:49, time: 0.447, data_time: 0.084, memory: 927, loss_rpn_cls: nan, loss_rpn_bbox: nan, loss_cls: nan, acc: 100.0000, loss_bbox: nan, loss: nan

> 2020-07-13 08:38:05,479 - mmdet - INFO - Epoch [8][100/1501]	lr: 2.000e-02, eta: 1:01:22, time: 0.410, data_time: 0.048, memory: 927, loss_rpn_cls: nan, loss_rpn_bbox: nan, loss_cls: nan, acc: 100.0000, loss_bbox: nan, loss: nan

probably due to loss becomin Nan!
any thoughts?!

**Comment by MolianWH** [comment]
@farshid-mi 
I also meet it. And it may caused by gradient explosion or gradient disappeared. You can try these ways:
1. Add grad_clip
2. Reduce learning rate
3. Change loss function
4. Increase warmup_iters
5. Reduce batch size 
6. Train on more gpus.
...

**Comment by hongrui16** [comment]
I got the same error,  when the epoch is smaller than 4, the evaluation can be implemented.
After the epoch increased to 5, the loss wan nan and got error, ERROR - The testing results of the whole dataset is empty, during evaluation.

**Comment by edwardnguyen1705** [comment]
@hongrui16 ,
I meet the same problem.

**Comment by hongrui16** [comment]
@edwardnguyen1705 
yep, decrease the learning rate can be useful to solve the error of loss lan

**Comment by dream-in-night** [comment]
i have meet the same problem, i slove it by create another conda enviroment.
the probleme come when i use my 1060 gpu, and if i run the same code in colab, the problem disappared .

**Comment by 654140593** [comment]
Dear Open-mmlab,

I had the same problem while testing convnext on my GTX1660ti. I was running this in command
python tools/test.py configs/convnext/cascade_mask_rcnn_convnext-s_p4_w7_fpn_giou_4conv1f_fp16_ms-crop_3x_coco.py checkpoints/cascade_mask_rcnn_convnext-s_p4_w7_fpn_giou_4conv1f_fp16_ms-crop_3x_coco_20220510_201004-3d24f5a4.pth --eval bbox

The output is like:
load checkpoint from local path: checkpoints/cascade_mask_rcnn_convnext-s_p4_w7_fpn_giou_4conv1f_fp16_ms-crop_3x_coco_20220510_201004-3d24f5a4.pth
[>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>] 5000/5000, 1.0 task/s, elapsed: 4996s, ETA:     0s
Evaluating bbox...
Loading and preparing results...
The testing results of the whole dataset is empty.
OrderedDict()

I ran several other models and they were OK but convnext does not work.
Can somebody help me with this problem? Any information could helpl.

All the best
Neil
